src/Form/TimesheetEditForm.php

```diff
@@ -266,11 +266,30 @@ function (FormEvent $event) {
                 if ($time === null) {
                     throw new \Exception('Cannot work with timesheets without start time');
                 }
-                $newEnd = clone $time;
-                $newEnd->setTime($end->format('H'), $end->format('i'));
 
-                if ($newEnd < $time) {
-                    $newEnd->modify('+ 1 day');
+                // Use duration to determine the actual end date
+                $duration = $timesheet->getDuration();
+                if ($duration !== null && $duration > 0) {
+                    // Calculate expected end time based on duration
+                    $expectedEnd = clone $time;
+                    $expectedEnd->modify('+ ' . $duration . ' seconds');
+
+                    // Create end time using the date from expected end time
+                    $newEnd = new \DateTime($expectedEnd->format('Y-m-d') . ' ' . $end->format('H:i:s'), new \DateTimeZone($expectedEnd->getTimezone()->getName()));
```

Comment:
I dont't understand that line. Why do use `$end->format('H:i:s')` instead of the calculated `expectedEnd` ?